app/assets/javascripts/app/modules/personal-key-formatter.js

```diff
@@ -0,0 +1,45 @@
+import { DelimitedTextFormatter } from 'field-kit';
+
+/**
+ * @const
+ * @private
+ */
+const DIGITS_PATTERN = /^[a-zA-Z0-9]*$/;
+
+/**
+ * @extends DelimitedTextFormatter
+ */
+class PersonalKeyFormatter extends DelimitedTextFormatter {
+  constructor() {
+    super('-', true);
+    this.maximumLength = 16 + 3;
+  }
+
+  /**
+   * @param {number} index
+   * @returns {boolean}
+   */
+
+  /* eslint-disable class-methods-use-this */
+  hasDelimiterAtIndex(index) {
+    return index === 4 || index === 9 || index === 14;
```

Comment:
We could expose num words and word length as `data-` attributes and read them in?